src/foam/strategy/BasicStrategizer.js

```diff
@@ -50,6 +50,10 @@ foam.CLASS({
               )
             );
 
+        if ( strategyPredicate != null  ) {
```

Comment:
The code above to build the predicate should be put in the 'else' block of this 'if', since there is no reason to do the above work if you aren't going to use it in this case.